### PR TITLE
[IMP] test_pylint: Add feature to validate expected errors vs a environ variable

### PR DESCRIPTION
**Pull request by moylop260**
This PR depend of https://github.com/OCA/maintainer-quality-tools/pull/229
You can view the real changes of this PR in the last commit diff.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
:+1: when #229 is merged

File: travis/test_pylint [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
+pylint_rcfile = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    'cfg',
+    "travis_run_pylint.cfg")
+count_errors = run_pylint.main([
+    "--config-file=" + pylint_rcfile,
+], standalone_mode=False)
+
+expected_errors = int(
+    os.environ.get('PYLINT_EXPECTED_ERRORS', 0))
```
Comment (moylop260) [inline travis/test_pylint]: FYI We are processing this fails here to have compatibility of this feature with `git hook`

**Comment by yvaucher** [comment]
:+1:

**Comment by nhomar** [comment]
:+1:

**Comment by pedrobaeza** [comment]
@guewen @lepistone Do you have any word about this or can I merge?

**Comment by pedrobaeza** [comment]
@moylop260, please rebase first

**Comment by guewen** [comment]
I'll look tomorrow

**Comment by moylop260** [comment]
@pedrobaeza 

@guewen is talking about the next PR:
> Posted on the wrong thread before; so I post again here:
https://github.com/OCA/maintainer-quality-tools/pull/232#issuecomment-133044182

This PR enable pylint test (don't affect current functionallity) and have 3 approvals then IMHO should be merged.

**Comment by moylop260** [comment]
Rebased

**Comment by guewen** [comment]
What is the use case for validating a build based on a number of errors?
If you have errors shouldn't you clean them? What if you have 17 errors, remove one of them because you deleted some code and introduce a new one, you won't see it?

**Comment by guewen** [comment]
Without an explanation of the *why* I'm :-1:

**Comment by moylop260** [comment]
@guewen 
This is the same case of 
https://github.com/OCA/maintainer-quality-tools/blob/master/.travis.yml#L29

 - INCLUDE="broken_module" SERVER_EXPECTED_ERRORS="1"  # test errors are detected

This feature is in core now.
Why was added?
The answer of this reply your current question.

**Comment by guewen** [comment]
> @guewen is talking about the next PR:

To be clear: I was talking about the whole set of new pylint PRs which make me uneasy.

I was not aware of `SERVER_EXPECTED_ERRORS`. And no, showing me that something similar exists does not respond to my questions, as I ask then the same questions for the existing code.

**Comment by moylop260** [comment]
As I told in next comment
https://github.com/OCA/maintainer-quality-tools/pull/230#issuecomment-133109569
You are talking about other PR.
You can comment in other one and I can gladly answer

Please, Can you unblock this one?

**Comment by moylop260** [comment]
I share my analogies of just this PR (and just this PR):
 - The `SERVER_EXPECTED_ERRORS` was added to enable the feature of test that `test_server` script is working fine with the travis builds.
   - The `PYLINT_EXPECTED_ERRORS` was added to enable the feature of test that `test_pylint` script is working fine with the travis builds.
 - The `broken_module` was added to force `test_server` errors and catch it to validate that is working fine.
    - The `broken_module` was modified to force `test_pylint` errors and catch it to validate that is working fine.

We had various attemp before.

 - 1. https://github.com/OCA/maintainer-quality-tools/pull/228/files Here just we can review if was exit(0) or exit(1) in this new PR we can verify that exactly the number of errors was found.

**Comment by moylop260** [comment]
@guewen 
Sorry, If the progressive PR generate confusion.
Can I help to explain another point of PR before of merge?

**Comment by guewen** [comment]
I finally got it, this is the self-test ability. I thought it was meant to be used for the normal builds.
Thanks for the details.

**Comment by moylop260** [comment]
@guewen
Thanks for your comments
Sorry by my unclear english.
Then merging...

**Comment by guewen** [comment]
I lift my :-1: 
Thanks for the details

**Pull request merged**